packaging/dependencies/amazon-cloudwatch-agent-ctl

```diff
@@ -158,7 +158,7 @@ agent_stop() {
      fi
 
      if [ "${SYSTEMD}" = 'true' ]; then
-          service "${agent_name}" stop || return
```

Comment:
Which operating systems rely on `service`? Do we still officially support those versions?

Reply:
To confirm, is the question which of our officially supported operating systems rely _solely_ on `service` and not on systemd's `systemctl` out of the box?

Reply:
Solely

Reply:
Got it. I could check but I don't think those would be impacted by this change? The current code path for the line being edited is only if `"${SYSTEMD}" = 'true'`.
So those operating systems if any are likely already using the else path and will continue to do so.

Reply:
But the check looks for upstart (the old one) first https://github.com/aws/amazon-cloudwatch-agent/blob/a98d8a2e1040ae8c3c0dd80d367da0486651a8d3/packaging/dependencies/amazon-cloudwatch-agent-ctl#L480-L483 unless I'm misconstruing upstart with something else.

Reply:
Thats correct. However when systemd is the actual service manager configured, /sbin/init is a symlink to systemd even though upstart's service command could be available.
```
$ ls -ltr /sbin/init
lrwxrwxrwx 1 root root 22 Jul 21 19:46 /sbin/init -> ../lib/systemd/systemd
$ which service
/usr/sbin/service
$ /sbin/init --version 2>/dev/null | grep -c upstart
0
``` 

But better to be safe so would the best way to go about this be: For the linux distributions in the list of supported operating systems [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Install-CloudWatch-Agent.html), I spin up an ec2 and confirm if /sbin/init is pointing to upstart or systemd?

Reply:
That's what I would do, yeah

Reply:
| OS | Is systemd default? | Is stop command recogninzed? |
| --- | --- | --- |
| Amazon Linux version 2014.03.02 or later | No | Yes |
| Amazon Linux 2 | Yes | N/A |
| Ubuntu Server versions 20.04, 18.04, 16.04 | Yes | N/A |
| Ubuntu Server version 14.04 | No | Yes |
| CentOS version 8 stream and version 7 | Yes | N/A |
| Red Hat Enterprise Linux (RHEL) versions 8, 7.7, 7.6, 7.5, 7.4, 7.2, and 7.0 | Yes | N/A |
| Debian version 10 | Yes | N/A |
| SUSE Linux Enterprise Server (SLES) version 15 and version 12 | Yes | N/A |
| Oracle Linux versions 7.8, 7.6, and 7.5 | Yes | N/A |

I also came across https://en.wikipedia.org/wiki/Systemd#Adoption which gives a good overview of when systemd was released as a default for an OS.

Keeping the following in mind, https://github.com/aws/amazon-cloudwatch-agent/blob/a98d8a2e1040ae8c3c0dd80d367da0486651a8d3/packaging/dependencies/amazon-cloudwatch-agent-ctl#L480-L483 if I were to summarize the impact:
* Though it checks for upstart first in the `if` clause, for OSes that have systemd as their default, /sbin/init is a symlink to systemd and hence the `if` clause would evaluate to false ===> For OSes with systemd as their default, `SYSTEMD` will be true.
* For OSes where systemd is not their default, the `if` clause will evaluate to true and `SYSTEMD` will be false. So the `stop` command (i.e. /sbin/stop) would be used instead.
* In the table above, I verified that either systemd is the default and for the few where it isnt, confirmed that the stop command is recognized.